gubernator.go

```diff
@@ -359,15 +359,18 @@ func (s *Instance) SetPeers(peers []PeerInfo) {
 		if peerInfo == nil {
 			shutdownPeers = append(shutdownPeers, p)
 			wg.Add(1)
-			go func() {
-				err := p.Shutdown(ctx)
-				if err != nil {
-					log.WithError(err).WithField("peer", p).Error("while shutting down peer")
-				}
-			}()
 		}
 	}
 
+	for _, p := range shutdownPeers {
+        go func() {
+            err := p.Shutdown(ctx, &wg)
+            if err != nil {
+                log.WithError(err).WithField("peer", p).Error("while shutting down peer")
+            }
```

Comment:
the wg.Done() should go on the next line, after this `}` . No need to pass the `wg` into `Shutdown()` right? This is the only place `Shutdown()` is called inside a go routine.

Reply:
Thanks. I made the suggested code change. I only checked that it compiled. To be honest I'm not sure when I have time to test this as I destroyed my entire test setup just before the Christmas holidays.

We were running a POC with gubernator as an nginx sidecar using the ngix helm chart with some lua code snippets and ingress annotations. This was successful although we also ran into https://github.com/mailgun/gubernator/issues/23 which is now also closed I see. We are now running some other throttling POCs.

Let me know if you think it is safe to pull this request as is. Its also fine by me if you want to incorporate this change in another branch if that is more convenient.